desktop/core/requirements.txt

```diff
@@ -9,7 +9,7 @@ channels-redis==2.3.2
 configobj==5.0.6
 cryptography==3.2
 django-auth-ldap==1.3.0
-Django==1.11.29 # Django 2 then 3?
+Django==2.0.13 # Django 2 then 3?
```

Comment:
One thing to check to:

from fresh install, 

`.../pip3 freeze` > pre.txt

`.../pip3 install Django==2.0.13`

`.../pip3 freeze` > post.txt

And check if upgtading django also bumped some of the modules? This might explain the behavior change.

Reply:
[post.txt](https://github.com/cloudera/hue/files/5807072/post.txt)
[pre.txt](https://github.com/cloudera/hue/files/5807073/pre.txt)
There is no difference (other than the Django version) between these two files.

Reply:
Ok, so last idea is just to put some `print()` and look if it gets encoded twice?

Reply:
I put print @ https://github.com/cloudera/hue/blob/fa87482247d04982fbeb3b552577c0a8a20dfb3f/apps/beeswax/src/beeswax/models.py#L542. This is confusing, output is in ss.
<img width="1792" alt="Screenshot 2021-01-15 at 3 42 37 PM" src="https://user-images.githubusercontent.com/36241930/104712377-7957ef80-5748-11eb-844b-4da7acb53d22.png">
<img width="1396" alt="Screenshot 2021-01-15 at 3 43 38 PM" src="https://user-images.githubusercontent.com/36241930/104712384-7d840d00-5748-11eb-92d2-4d2b7171d527.png">

Reply:
looks like for the same input it is converting in a different way.
And what I found is that x = b'1' and base64.b64decode(x) =  b'MQ=='
and when x = b"b'MQ=='" and base64.b64decode(x) = b'l\xc4'
So not know where is getting changed in another encoded string.

Reply:
One more thing for this test https://github.com/cloudera/hue/blob/fa87482247d04982fbeb3b552577c0a8a20dfb3f/apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py#L233 we are calling 3 time `client.get_table` in the given test. so if commented out 2 get_table commands then this test is getting passed.
So I think after calling the first time i.e (https://github.com/cloudera/hue/blob/fa87482247d04982fbeb3b552577c0a8a20dfb3f/apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py#L277) secret and guid are getting encoded.

Reply:
Nice progress!

Just it still does not explain why it would start failing because of Django. 
Either we would need to reset the handle to the initial value in the tests but this is not supposed to happen as it was immutable by design.

Did you track if it leak or is now getting persisted by the ORM? You could print the values of the handles / secrets to debug
https://github.com/cloudera/hue/blob/master/apps/beeswax/src/beeswax/server/hive_server2_lib.py#L772 get_handle()

Reply:
looks like `secret` and `guid` are in string form here https://github.com/cloudera/hue/blob/master/apps/beeswax/src/beeswax/models.py#L464. I think this is a problem, but not able to go much deeper such that it can be solved. @romainr

Reply:
https://docs.djangoproject.com/en/2.0/_modules/django/db/models/fields/#TextField

Are we storing bytes instead of unicode? (in Python 3 str means unicode)

https://stackoverflow.com/questions/18034272/python-str-vs-unicode-types


I quicked checked but probably missed it, but any documentation about a change with TextField in Django 2?

Reply:
Didn't find any documentation related to TextField and Django2.

Reply:
> Nice progress!
> 
> Just it still does not explain why it would start failing because of Django.
> Either we would need to reset the handle to the initial value in the tests but this is not supposed to happen as it was immutable by design.
> 
> Did you track if it leak or is now getting persisted by the ORM? You could print the values of the handles / secrets to debug
> https://github.com/cloudera/hue/blob/master/apps/beeswax/src/beeswax/server/hive_server2_lib.py#L772 get_handle()

Didn't find any leak.